### DISCUSSION TITLE
Is it possible to query remote API "per row"?

### DISCUSSION THREAD

**Discussion by dolzenko** [duckdb/duckdb#14845] [General]
Assuming I need to enrich my data with the result of some API call and then do GROUP BY on that result, how should I go about it? Is it a sensible use case? Currently I 
1. load my data (in Go) from  parquet files on S3
2. then for each row I make an API call (parallelized a lot, like 100s of threads)
3. then do the rollup inside the app

It seems like 1/ and 3/ can be easily migrated to DuckDB and save me few lines of Go, but what about 2/?

**Comment by Maxxen** [comment]
The [httpclient community extension](https://duckdb.org/community_extensions/extensions/http_client.html) might help with this.
